docs/source/en/basics/router.md

```diff
@@ -0,0 +1,328 @@
+
+exports.edit = function* () {};
+
+exports.update = function* () {};
+
+exports.destroy = function* () {};
+```
+
+Methods that are not needed may not be implemented in `posts.js` and the related URL paths will not be registered to Router neither.
+
+## destroy in action
+
+The destroy syntax will demonstrated by more examples.
+
+#### Acquiring Parameters
+
+#### via Query String
+
+```js
+// app/router.js
+module.exports = app => {
+  app.get('/search', 'search');
+};
+
+// app/controller/search.js
+module.exports = function* (ctx) {
+  ctx.body = `search: ${this.query.name}`;
+};
+
+// curl http://127.0.0.1:7001/search?name=egg
+```
+
+#### via Named Parameters
+
+```js
+// app/router.js
+module.exports = app => {
+  app.get('/user/:id/:name', 'user.info');
+};
+
+// app/controller/user.js
+exports.info = function* (ctx) {
+  ctx.body = `user: ${ctx.params.id}, ${ctx.params.name}`;
+};
+
+// curl http://127.0.0.1:7001/user/123/xiaoming
+```
+
+#### acquiring complex parameters
+
+Regular expressions, as well, can be used in routing rules to aquire parameters more flexible:
+
+```js
+// app/router.js
+module.exports = app => {
+  app.get(/^\/package\/([\w-.]+\/[\w-.]+)$/, 'package.detail');
+};
+
+// app/controller/package.js
+exports.detail = function* (ctx) {
+  // If the request URL is matched by the regular expression, parameters can be get from ctx.params according to the capture group orders.
+  // For the user request below, for example, the value of `ctx.params[0]` is `egg/1.0.0`
+  ctx.body = `package:${ctx.params[0]}`;
+};
+
+// curl http://127.0.0.1:7001/package/egg/1.0.0
+```
+
+### Acquiring Form Contents
+
+```js
+// app/router.js
+module.exports = app => {
+  app.post('/form', 'form');
+};
+
+// app/controller/form.js
+module.exports = function* (ctx) {
+  ctx.body = `body: ${JSON.stringify(ctx.request.body)}`;
+};
+
+// simulate a post request.
+// curl -X POST http://127.0.0.1:7001/form --data '{"name":"controller"}' --header 'Content-Type:application/json'
+```
+
+> P.S.:
+
+> If you perform a POST request directly, an **error** will occur: 'secret is missing'. This error message comes from [koa-csrf/index.js#L69](https://github.com/koajs/csrf/blob/2.5.0/index.js#L69).
+
+> **Reason**: the framework verifies the CSFR value specially for form POST requests, so please submit the CSRF key as well when you submit a form. For more detail refer to [Keep Away from CSRF Threat](https://eggjs.org/zh-cn/core/security.html#å®‰å…¨å¨èƒcsrfçš„é˜²èŒƒ)
+
+> **Note**: the verification is performed because the framework builds in a security plugin [egg-security](https://github.com/eggjs/egg-security) that provides some default security practices and this plugin is enabled by default. In case you want to disable some security protections, just configure the enable attribute of if to be false.
+
+> "Unless you know the consequence consciously, it's recommended not to disable functions provided by the security plugin"
```

Comment:
‘unless ...’
suggestion: 'Unless you clearly confirm ..., it's not recommended ...'